tests/unit_tests/test_lobbyconnection.py

```diff
@@ -255,3 +256,230 @@ def test_avatar_select_no_avatar(mock_query, fa_server_thread):
     with pytest.raises(KeyError):
         fa_server_thread.command_avatar({'action': 'select'})
 
+# handle action
+def test_handle_action_ping(fa_server_thread):
+    fa_server_thread.sendReply = mock.Mock()
+    fa_server_thread.handleAction('PING', mock.Mock())
+    fa_server_thread.sendReply.assert_called_once_with('PONG')
+
+def test_handle_action_pong(fa_server_thread):
+    assert fa_server_thread.ponged == False
+    fa_server_thread.handleAction('PONG', mock.Mock())
+    assert fa_server_thread.ponged == True
+
+def test_handle_action_faclosed(fa_server_thread):
+    fa_server_thread.player = mock.Mock()
+    fa_server_thread.handleAction('FA_CLOSED', mock.Mock())
+    fa_server_thread.player.setAction.assert_called_once_with('NOTHING')
+    fa_server_thread.player.gameThread.abort.assert_called_once_with()
+
+def test_handle_action_unknownData(fa_server_thread):
+    fa_server_thread.receiveJSON = mock.Mock()
+    stream = mock.Mock()
+    fa_server_thread.handleAction('CrazyThing', stream)
+    fa_server_thread.receiveJSON.assert_called_once_with(\
+        'CrazyThing', stream)
```

Comment:
THIS IS IMPORTANT:
over this way we can execute commands over the Stream API:
https://github.com/FAForever/server/blob/develop/src/lobbyconnection.py#L2413